iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesParser.java

```diff
@@ -49,18 +49,26 @@ public class KubernetesParser extends YamlParser {
 
   private final HelmProcessor helmProcessor;
   private final LocationShifter locationShifter;
+  private final KubernetesParserStatistics kubernetesParserStatistics;
 
-  public KubernetesParser(HelmProcessor helmProcessor, LocationShifter locationShifter) {
+  public KubernetesParser(HelmProcessor helmProcessor, LocationShifter locationShifter, KubernetesParserStatistics kubernetesParserStatistics) {
     this.helmProcessor = helmProcessor;
     this.locationShifter = locationShifter;
+    this.kubernetesParserStatistics = kubernetesParserStatistics;
   }
 
   @Override
   public FileTree parse(String source, @Nullable InputFileContext inputFileContext) {
     if (!hasHelmContent(source)) {
-      return super.parse(source, inputFileContext);
+      kubernetesParserStatistics.incrementPureKubernetesFileCount();
+      var result = super.parse(source, inputFileContext);
```

Comment:
Maybe instead methods like `recordPureKubernetesFile` in `KubernetesParserStatistics`:
```java
public <T> T recordPureKubernetesFile(Supplier<T> callable) {
  incrementPureKubernetesFileCount();
  var result = callable();
  incrementPureKubernetesParsedFileCount();
  return result;
}
```
and use it here like
```suggestion
return kubernetesParserStatistics.recordPureKubernetesFile(() -> super.parse(source, inputFileContext))`
```
?

Reply:
Good point